src/components/collapsible-section-stateful/collapsible-section-stateful.tsx

```diff
@@ -0,0 +1,115 @@
+import { Component, Event, EventEmitter, h, Prop, Watch } from '@stencil/core';
+import { Action } from '../collapsible-section/action';
+import { getState, setState } from '../../util/state-service';
+
+/**
+ * @slot - Content to put inside the collapsible section
+ * @exampleComponent limel-example-collapsible-section-stateful
+ */
+@Component({
+    tag: 'limel-collapsible-section-stateful',
+    shadow: true,
+})
+export class CollapsibleSectionStateful {
```

Comment:
> How do you feel about a separate component for this vs just a property on the regular component?
> 
> If we add other stateful components, should we create separate components for those as well? It feels like we might end up with duplicates for a lot components with one stateful and one regular component which might feel weird for the consumer when reading the documentation etc.
> 
> Another problem with this approach is that it creates some duplication. If we add a new property to the regular component, we need to remember to add it to the stateful one as well and remember to map it to the inner component.

Oh, sorry, I thought you wanted it as a separate component, so it wouldn't increase complexity of the original component.

I don't know, I guess I can see reasonable arguments either way…

It feels like the idea for setting a stateKey automatically when using limel-form is looking less likely to happen, and I think that if we involve the consumer in that part, we're not really gaining much "for free" from the consumer's perspective by doing the state handling inside lime-elements, so I'm thinking we might scrap this in the end. So I think I'll convert this to a draft and put it on ice until we know more about how we want the whole thing to work. It might not be the best issue to spend a lot of time on in this cycle. We can probably get more value in less time by focusing on other issues.

The first commit, that just adds tests and stuff to the original collapsible, I'd still like to have reviewed and merged though 😊